webapp/channels/src/components/integrations/abstract_outgoing_webhook.tsx

```diff
@@ -568,7 +590,7 @@ export default class AbstractOutgoingWebhook extends React.PureComponent {
                                 className='btn btn-primary'
                                 type='submit'
                                 spinning={this.state.saving}
-                                spinningText={localizeMessage(this.props.loading.id, this.props.loading.defaultMessage)}
+                                spinningText={localizeMessage(this.props.loading.id as string, this.props.loading.defaultMessage as string)}
```

Comment:
Not a big fan of the assertions, but unless @M-ZubairAhmed says otherwise, let's leave it like this.

The only solution I can imagine is to create a new type for MessageDescriptor that goes more or less like this:

```
type requiredMessageDescriptor = Required<Pick<MessageDescriptor, 'id' | 'defaultMessage'>> & Omit<MessageDescriptor, 'id' | 'defaultMessage'>
```

But that seems a too convoluted type and we would have to do a stronger refactoring to use the correct one in many places probably.

Reply:
Agreed, I had thought something similar but didn't want to be messing with the typing across more of the codebase. If we want to make changes here though just let me know.

Reply:
yes lets leave it as it is as @larkox suggested. I usually end up removing these custom localization with react-intl API in my PRs. But this is fine